### ISSUE TITLE
Elasticsearch doesn't work on one of elassandra nodes

### ISSUE THREAD

**Issue by arturalbov** [strapdata/elassandra#211]
**Elasticsearch version**: 5.5.0.19

**Plugins installed**: []

**JVM version** (`java -version`): 1.8.0_162

**OS version** (`uname -a` if on a Unix-like system): Linux elassandra-1 4.14.22+ #1 SMP Thu May 10 17:54:42 PDT 2018 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:

Elasticsearch doesn't work on one of elassandra nodes. We got some warnings in logs:
```
2018-07-10 11:34:34,072 WARN  [elasticsearch[10.12.44.12][clusterService#updateTask][T#1]] org.elasticsearch.cluster.routing.IndexRoutingTable$Builder.&lt;init&gt;(IndexRoutingTable.java:380) Keyspace not available for index [[bitcoin.contract_summary/4jrW33ttTA-ZGdXBuNbNWg]]
```
Full logs [logs-from-elassandra-in-elassandra-0-2.txt](https://github.com/strapdata/elassandra/files/2180180/logs-from-elassandra-in-elassandra-0-2.txt)

Elastic node seems to be healthy but doesn't well on requests (examples below). Cassandra works well on this node.

Cluster health:
```
root@elassandra-0:/# curl 'localhost:9200/_cluster/health?pretty'
{
  "cluster_name" : "CYBER_SEARCH",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 5,
  "number_of_data_nodes" : 5,
  "active_primary_shards" : 0,
  "active_shards" : 0,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 100.0
}
```
Cluster health from the good node:
```
root@elassandra-1:/# curl 'localhost:9200/_cluster/health?pretty'
{
  "cluster_name" : "CYBER_SEARCH",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 5,
  "number_of_data_nodes" : 5,
  "active_primary_shards" : 35,
  "active_shards" : 35,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 100.0
}
```

Stats:
```
root@elassandra-0:/# curl 'localhost:9200/_stats?pretty'
{
  "_shards" : {
    "total" : 0,
    "successful" : 0,
    "failed" : 0
  },
  "_all" : {
    "primaries" : { },
    "total" : { }
  },
  "indices" : { }
}
```

Search:
```
root@elassandra-0:/# curl 'localhost:9200/_search?q=42'
{"took":0,"timed_out":false,"_shards":{"total":0,"successful":0,"failed":0},"hits":{"total":0,"max_score":0.0,"hits":[]}}
```

Nodes stats:
[nodes-stats.txt](https://github.com/strapdata/elassandra/files/2180196/nodes-stats.txt)

This behavior appears periodically after restarting node (sometimes few times in a row).

**Comment by vroyer** [comment]
Hi,
It seems that you had a network or ressources issue, because according to your logs, some node are not yet visible.

```
2018-07-10 10:32:17,612 INFO  [GossipTasks:1] org.apache.cassandra.gms.Gossiper.doStatusCheck(Gossiper.java:812) FatClient /10.12.46.11 has been silent for 30000ms, removing from gossip
2018-07-10 10:32:24,615 INFO  [GossipTasks:1] org.apache.cassandra.gms.Gossiper.doStatusCheck(Gossiper.java:812) FatClient /10.12.48.9 has been silent for 30000ms, removing from gossip
2018-07-10 10:32:46,623 INFO  [GossipTasks:1] org.apache.cassandra.gms.Gossiper.doStatusCheck(Gossiper.java:812) FatClient /10.12.47.9 has been silent for 30000ms, removing from gossip
```

Moreover, this has cause a race condition when computing the routing table causing these exceptions in your logs. This is fixed and will be released in the next version 5.5.0.20.

```
2018-07-10 10:31:54,662 WARN  [elasticsearch[10.12.46.12][clusterService#updateTask][T#1]] org.elasticsearch.cluster.routing.IndexRoutingTable$Builder.<init>(IndexRoutingTable.java:380) Keyspace not available for index [[bitcoin.tx/jxO7uBfsRrmBkd0vrjFo9Q]]
java.lang.AssertionError: recovery source only available on unassigned or initializing shard but was STARTED
	at org.elasticsearch.cluster.routing.ShardRouting.<init>(ShardRouting.java:117)
	at org.elasticsearch.cluster.routing.ShardRouting.<init>(ShardRouting.java:83)
```

**Comment by arturalbov** [comment]
The point is those IP addresses in logs relate to old nodes.
We're using k8s in GCE cluster and after I restart all nodes they changing their IP addresses. So, addresses that you've mentioned in logs are old addresses of nodes.

By the way, when you're planning 5.5.0.20 release?

**Comment by vroyer** [comment]
Fix for k8s ip address replacement is included in release 5.5.0.20 and 6.2.3.3.
Thanks for your feedback on running elassandra in k8s GCP.

**Comment by arturalbov** [comment]
Nice! Thanks
